src/app/qgsguivectorlayertools.h

```diff
@@ -71,8 +71,20 @@ class QgsGuiVectorLayerTools : public QObject, public QgsVectorLayerTools
      */
     bool saveEdits( QgsVectorLayer* layer ) const override;
 
+    /**
+     * Copy and move features with defined translation.
+     *
+     * @param layer The layer
+     * @param ids The IDs of the features to be moved
+     * @param dx The translation on x
+     * @param dy The translation on y
+     * @return True if successful.
+     */
+    bool copyMoveFeatures( QgsVectorLayer* layer, QgsFeatureIds ids, double  dx = 0, double dy = 0 ) const override;
```

Comment:
May be more flexible to accept a QgsFeatureRequest instead of QgsFeatureIds, but no strong opinion from me.